code/src/cljs/sixsq/nuvla/ui/edges/views.cljs

```diff
@@ -779,13 +776,139 @@
         :i-per-page-multipliers [1 2 4]}])))
 
 
+(defn- get-name-as-keyword
+  [tr q-key]
+  (tr [(-> q-key name keyword)]))
+
+(defn- TagsEditModeRadio
+  [edit-mode opened-modal change-mode]
+  (let [tr               (subscribe [::i18n-subs/tr])
+        active?          (= opened-modal edit-mode)
+        font-weight      (if active? 700 400)
+        setting-tags?    (= spec/modal-tags-set-id edit-mode)]
+    [ui/Radio {:style     {:font-weight font-weight}
+               :label     (str (get-name-as-keyword @tr edit-mode)
+                               (when setting-tags? (str " (" (@tr [:tags-overwrite]) "!)")))
+               :checked   active?
+               :on-change #(change-mode edit-mode)}]))
+
+
+(defn- ButtonAskingForConfirmation
+  [_form-tags close-fn]
+  (let [tr               (subscribe [::i18n-subs/tr])
+        selected-count   (subscribe [::subs/selected-count ::spec/select])
+        edit-mode        (subscribe [::subs/opened-modal])
+        mode             (r/atom :idle)
+        edit-mode->color {spec/modal-tags-add-id     :green
+                          spec/modal-tags-remove-all :red
+                          spec/modal-tags-remove-id  :red
+                          spec/modal-tags-set-id     :red}]
+    (fn [form-tags _close-fn]
+      (let [text      (get-name-as-keyword @tr @edit-mode)
+            call-back close-fn
+            update-fn (fn []
+                        (dispatch [::events/update-tags
+                                   @edit-mode
+                                   {:tags         form-tags
+                                    :call-back-fn call-back
+                                    :text text}]))
+            disabled? (or (zero? @selected-count)
+                          (and  (not= spec/modal-tags-remove-all @edit-mode)
+                                (zero? (count form-tags))))]
+        (if (= :idle @mode)
+          [:div
+           [:span (str text "?")]
+           [ui/Button {:aria-label "edit tags"
+                       :color    (edit-mode->color @edit-mode)
+                       :disabled disabled?
+                       :active   true
+                       :style    {:margin-left "2rem"}
+                       :on-click (fn [] (reset! mode :confirming))}
+            [uix/Icon {:style {:margin 0}
+                       :name "fa-check"}]]]
+          [:div
+           [:span "Are you sure? " ]
+           [uix/Button {:text     (str "Yes, " text)
+                        :disabled disabled?
+                        :color    (edit-mode->color @edit-mode)
+                        :on-click update-fn}]
+           [ui/Button {:aria-label "do not confirm editing tags"
+                       :on-click (fn [] (reset! mode :idle))}
+            [ui/Icon {:style {:margin 0}
+                      :name "fa-xmark"}]]])))))
+
+(defn BulkUpdateModal
```

Comment:
Bulk edit should be possible for many resources. To be extracted and make it reusable. Can be done later.